clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -659,6 +655,12 @@ else if (error == Errors.CLUSTER_AUTHORIZATION_FAILED)
             this.accumulator.unmutePartition(batch.topicPartition);
     }
 
+    private String formatErrMsg(ProduceResponse.PartitionResponse response) {
```

Comment:
I'm wondering if we should print an error message even if one is not provided - we could get the default error message from the `ApiException` linked to the `Errors`. An example would be for the OUT_OF_ORDER_SEQUENCE_NUMBER case:

Without the change:
```
[2020-11-12 11:26:26,775] WARN Got error produce response with correlation id 2 on topic-partition test-0, retrying (2147483646 attempts left). Error: OUT_OF_ORDER_SEQUENCE_NUMBER.(org.apache.kafka.clients.producer.internals.Sender:610)
```


```
[2020-11-12 11:26:26,775] WARN Got error produce response with correlation id 2 on topic-partition test-0, retrying (2147483646 attempts left). Error: OUT_OF_ORDER_SEQUENCE_NUMBER. Error Message: The broker received an out of order sequence number. (org.apache.kafka.clients.producer.internals.Sender:610)
```

Reply:
Indeed, we could. I am not sure that it brings much more information though so I am fine with keeping it as it is.